ideascube/mixins.py

```diff
@@ -72,10 +74,10 @@ def _set_available_tags(self, context):
 
     def get_context_data(self, **kwargs):
         context = super(FilterableViewMixin, self).get_context_data(**kwargs)
-        for key in ('q', 'kind', 'lang'):
+        for key in ('q', 'kind', 'lang', 'source'):
             context[key] = self.request.GET.get(key)
         context['tags'] = self.request.GET.getlist('tags')
-        current_filters = [(k, context[k]) for k in ('kind', 'lang')
+        current_filters = [(k, context[k]) for k in ('kind', 'lang', 'source')
```

Comment:
We do not set current_filters for `q`. It would complicate the for loop with a inner if.